lib/identity_pki/token.ex

```diff
@@ -18,16 +22,35 @@ defmodule IdentityPki.Token do
     encrypt_text(json, key)
   end
 
+  @spec open(String.t(), String.t()) :: {:ok, Map.t()} | {:error, atom()}
   def open(token, key \\ secret_key()) do
-    # TODO: check expires_at
-    # TODO: check HMAC
-    [text, iv, tag] =
-      String.split(token, "--")
-      |> Enum.map(&Base.decode64!/1)
+    with [text, iv, tag] <- String.split(token, "--"),
```

Comment:
**Suggestion:** I wasn't super familiar with the `with` syntax but it's really nifty! I wonder if it might be worth adding some comments to help clarify for those newer to Elixir, e.g. "In order for the code in the `do` block to execute, all of the chained clauses must match. If any of the them fail the `else` block is executed." and/or a link to the relevent [Elixir docs](https://hexdocs.pm/elixir/master/Kernel.SpecialForms.html#with/1).

What do you think?

Reply:
Yep, makes sense. I added `open` docs in 32e6aa18b41994e9ec17933b802bd42a661f66df, and more comprehensive docs to the module as a whole.